### ISSUE TITLE
Invalid parameter values in [dpqr] functions should return NaN

### ISSUE THREAD

**Issue by piklprado** [piklprado/sads#8]
Make invalid parameters return NaN, instead than stoping. Done in some functions, check in all.

**Comment by andrechalom** [comment]
I'm working on the 'dbs' function for a start. This function stops when:
1. some parameter is infinite
2. N or S is less or equal to 0
3. at least one x is less than 1
4. at least one x is not a whole number.

For what I see in the base [dpqr] functions, like 'dexp', the problems (1.) and (2.) should return NaN instead of stoping. I'm assuming that (3.) and (4.) should return NaN as well.

Commit https://github.com/andrechalom/sads/commit/9f7891040649f99d63dd2ef5144ba883248a04f0 implements this, as well as assuring that all NaN results issue a warning.

**Comment by andrechalom** [comment]
Zipf and Power distributions currently set distribution value to -Inf when !is.wholenumber. I'm assuming that this is a mistake, and setting it to NaN for coherence.

**Comment by andrechalom** [comment]
Just rewrote the [d] functions to produce NaN instead of stopping. As most [p] functions are based on cumsum(d..., ...), I wrote a cumsum wrapper to do most of the work. Partial results in https://github.com/andrechalom/sads/tree/Fix8

**Comment by andrechalom** [comment]
Now the pull request from this branch fixes both #8 and #35, as both issues are very closely related. I rewrote most of the code for the existing [dpq] functions, they all now return NaN for invalid parameters and have more organized and efficient code. 

One last issue addressed in this pull request is the substitution of approxfun by approx in all [q] functions. As exemplified below with code from *qrbs*, this gives a slight performance gain:

```R
> library(microbenchmark)
> N=100
> S=25
> Y <- 1:S
> X <- prbs(Y, N=N, S=S)
> p=0:100/100
> qapprox <- function() approx(x=X, y=Y, xout=p, method="constant", rule=2)$y
> qapproxfun <- function() {f <- approxfun(x=X, y=Y, method="constant", rule=2); f(p)}
> microbenchmark(qapprox(), qapproxfun())
Unit: microseconds
         expr    min      lq     mean median      uq     max neval cld
    qapprox() 41.002 42.3300 46.34797 44.032 47.1600 114.686   100  a 
 qapproxfun() 63.881 66.2835 73.25322 67.751 76.1545 202.390   100   b
```
